jaeger-production-template.yml

```diff
@@ -163,3 +169,68 @@ items:
     selector:
       jaeger-infra: query-pod
     type: LoadBalancer
+- apiVersion: apps/v1
+  kind: DaemonSet
+  metadata:
+    name: jaeger-agent
+    labels:
+      app: jaeger
+      jaeger-infra: agent-daemonset
+  spec:
+    selector:
+      matchLabels:
+        app: jaeger
+        jaeger-infra: agent-instance
+    template:
+      metadata:
+        labels:
+          app: jaeger
+          jaeger-infra: agent-instance
+      spec:
+        containers:
+        - name: agent-instance
+          image: jaegertracing/jaeger-agent:1.2
+          command:
+          - "/go/bin/agent-linux"
+          - "--config-file=/conf/agent.yaml"
+          volumeMounts:
+          - name: jaeger-configuration-volume
+            mountPath: /conf
+          ports:
+          - containerPort: 5775
+            protocol: UDP
+          - containerPort: 6831
+            protocol: UDP
+          - containerPort: 6832
+            protocol: UDP
+        volumes:
+          - configMap:
+              name: jaeger-configuration
+              items:
+                - key: agent
+                  path: agent.yaml
+            name: jaeger-configuration-volume
+- apiVersion: v1
```

Comment:
This service is unnecessary (as the pods on each node send their UDP reports to the host IP)

Reply:
True, but one benefit of having a service is that the instrumented application can refer to the agent via the hostname `jaeger-agent` and not care about whether it's being deployed as a daemonset or regular pod.

Reply:
This means traffic may route into different nodes - the whole point of running the daemonset was that each pod would be able to route to the (single) jaeger-agent on the node it's scheduled on.

Reply:
To reinforce @pieterlange comment, this means traffic **will** be **mostly** routed to a different nodes